core/src/main/java/org/axonframework/commandhandling/distributed/DistributedCommandBus.java

```diff
@@ -98,8 +99,9 @@ public DistributedCommandBus(CommandRouter commandRouter, CommandBusConnector co
     public <C> void dispatch(CommandMessage<C> command) {
         if (NoOpMessageMonitor.INSTANCE.equals(messageMonitor)) {
             CommandMessage<? extends C> interceptedCommand = intercept(command);
-            Member destination = commandRouter.findDestination(command)
-                    .orElseThrow(() -> new CommandDispatchException("No node known to accept " + command.getCommandName()));
+            Member destination = commandRouter.findDestination(interceptedCommand)
+                                              .orElseThrow(() -> new NoHandlerForCommandException(
+                                                      "No node known to accept " + interceptedCommand.getCommandName()));
```

Comment:
Nit: I always like to wrap the failing bit in square brackets to clarify the _thing_ which didn't work, so something like this:
`new NoHandlerForCommandException(String.format("No node known to accept [%s]", interceptedCommand.getCommandName()))`

Reply:
Adjusted.